watchtower-core/src/main/java/com/tracelink/appsec/watchtower/core/auth/controller/UserMgmtController.java

```diff
@@ -103,9 +103,9 @@ else if (roleEntity == null && role != -1L) {
 			if (role == -1) {
 				user.setRoles(new HashSet<>());
 			}
-			// or add the correct one
+			// or set the correct one
 			else {
-				user.getRoles().add(roleEntity);
+				user.setRoles(new HashSet<>(Arrays.asList(roleEntity)));
```

Comment:
Can a user only have one role?

Reply:
yes. The UI restricts this as well. This was a mistake left over from when I converted from the fixed Roles to Privileges.